lib/test.js

```diff
@@ -53,6 +53,57 @@ function plan(count) {
 	this.plan(count, captureStack(this.plan));
 }
 
+function tryTest(title, fn, ...args) {
+	if (typeof title === 'function') {
+		args.unshift(fn);
+		fn = title;
+		title = undefined;
+	}
+
+	let committed = false;
+	let discarded = false;
+
+	this.addPendingAttemptAssertion();
+
+	const running = this
+		.forAttempt(title, t => fn(t, ...args))
+		.run();
+
+	return running.then(ret => {
+		const {passed, error, title, logs} = ret;
+		const errors = error ? Array.isArray(error) ? [...error] : [error] : [];
+
+		return {
+			passed,
+			errors,
+			title,
+			logs,
+			commit: ({retainLogs = true} = {}) => {
+				if (committed) {
+					return;
+				}
+				if (discarded) {
+					this.saveFirstError(new Error('Can\'t commit the result that was previously discarded'));
+					return;
+				}
+				committed = true;
+				this.countAttemptAssertion({inc: 1, result: ret, retainLogs});
```

Comment:
This should be the number of assertions made in the attempt, not hardcoded to `1`.

Reply:
This value is not returned from the attempt. It can be obtained from `assertCount` private field of the attempt instance. Is it acceptable?